### DISCUSSION TITLE
Cannot find cloudflare iframe after the last update to the cf

### DISCUSSION THREAD

**Discussion by SaberTawfiq** [seleniumbase/SeleniumBase#2917] [General]
After the CF update today cannot find the iframe 
the code that checking for and still cannot find it 

### def click_turnstile_and_verify(sb, excel_file_path):
    try:
            # Wait for the page to fully load
            sb.wait_for_ready_state_complete(timeout=30)
            iframe_found = False
            for attempt in range(120):  # Check for the iframe every 2 seconds, up to 4 minutes
                print(f"Attempt {attempt + 1}: Checking for Cloudflare iframe...")
                if sb.is_element_present('iframe[src*="cloudflare"]'):
                    iframe_found = True
                    break
                time.sleep(2)
            if not iframe_found:
                current_time = datetime.now().strftime("%I:%M:%S %p")
                print(colorama.Fore.RED + f"{current_time} - Cloudflare iframe not found after 4 minutes.")
                return

            # Switch to the iframe containing the Cloudflare Turnstile challenge
            sb.switch_to_frame('iframe[src*="cloudflare"]')

            # Wait for the Cloudflare Turnstile element inside the iframe
            sb.wait_for_element_visible("#qrVk5 > div.cb-c[role='alert']", timeout=15)

            if sb.is_element_visible("#qrVk5 > div.cb-c[role='alert']"):
                current_time = datetime.now().strftime("%I:%M:%S %p")
                print(colorama.Fore.LIGHTWHITE_EX + f"{current_time} - CloudFlare Turnstile Element found.")

                # Wait for the checkbox to be present and visible
                sb.wait_for_element_present("#qrVk5 > div.cb-c[role='alert'] > label.cb-lb > input[type='checkbox']", timeout=15)
                sb.wait_for_element_visible("#qrVk5 > div.cb-c[role='alert'] > label.cb-lb > input[type='checkbox']", timeout=15)

                # Handle GUI interaction if needed
                sb.uc_gui_click_cf()

                # Perform any additional actions needed (e.g., clicking the checkbox)
                sb.click("#qrVk5 > div.cb-c[role='alert'] > label.cb-lb > input[type='checkbox']")

                # Wait for the success icon or confirmation
                sb.wait_for_element_visible("svg#success-icon", timeout=15)
                if sb.is_element_visible("svg#success-icon"):
                    current_time = datetime.now().strftime("%I:%M:%S %p")
                    print(colorama.Fore.LIGHTWHITE_EX + f"{current_time} - Turnstile verification succeeded.")
                else:
                    current_time = datetime.now().strftime("%I:%M:%S %p")
                    print(colorama.Fore.LIGHTCYAN_EX + f"{current_time} - Turnstile verification did not succeed.")

            else:
                current_time = datetime.now().strftime("%I:%M:%S %p")
                print(colorama.Fore.LIGHTCYAN_EX + f"{current_time} - CloudFlare Turnstile Element not found. Switching back...")

            # Switch back to the default content
            sb.driver.switch_to.default_content()

    except Exception as e:
        current_time = datetime.now().strftime("%I:%M:%S %p")
        print(colorama.Fore.RED + f"{current_time} - An error occurred: {e}")
        sb.driver.switch_to.default_content()

**Comment by SaberTawfiq** [comment]
https://visa.vfsglobal.com/egy/en/hrv/login

**Reply by misterpeople** [reply]
Were u able to do it in this website?
